Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityToolkitFiles.cs

```diff
@@ -59,15 +77,25 @@ public static void OnPostprocessAllAssets(string[] importedAssets, string[] dele
             {
                 searchForFoldersTask.Wait();
 
+                // Identify the unique set of imported folders instead of unique files
+                // in order to avoid duplicating per-folder work.
+                HashSet<string> importedFolders = new HashSet<string>();
                 foreach (string asset in importedAssets.Concat(movedAssets))
                 {
-                    string folder = ResolveFullAssetsPath(asset);
+                    string fullAssetPath = ResolveFullAssetsPath(asset);
+                    string folder = Path.GetDirectoryName(fullAssetPath);
+                    importedFolders.Add(folder);
+                }
+
+                foreach (string folder in importedFolders)
+                {
                     TryRegisterModuleFolder(folder);
                 }
 
                 foreach (string asset in deletedAssets.Concat(movedFromAssetPaths))
                 {
-                    string folder = ResolveFullAssetsPath(asset);
+                    string fullAssetPath = ResolveFullAssetsPath(asset);
```

Comment:
Could we just pattern match the sentinel files in this callback, so that if a folder is moved we don't do a deep recursive search for each file in that folder?

Reply:
Yep, makes sense!